File: Source/Frontend/src/app/core/dto/turnover-sheet-details-dto.ts
import { BillClassDto } from "./bill-class-dto";

export interface TurnoverSheetDetailsDto {
    name: string,
    incomingBalanceActive: number,
    incomingBalancePassive: number,
    turnoverDebit: number,
    turnoverCredit: number,
    outgoingBalanceActive: number,
    outgoingBalancePassive: number,
    billClasses: BillClassDto[]
}

File: Source/API/DTO/TurnoverSheetDetailsDTO.cs
namespace OCBManager.API.DTO
{
    public record TurnoverSheetDetailsDTO(
        string Name,
        decimal IncomingBalanceActive,
        decimal IncomingBalancePassive,
        decimal TurnoverDebit,
        decimal TurnoverCredit,
        decimal OutgoingBalanceActive,
        decimal OutgoingBalancePassive,
        List<BillClassDTO> BillClasses);

    public record BillClassDTO(
        int Id,
        string Name,
        decimal IncomingBalanceActive,
        decimal IncomingBalancePassive,
        decimal TurnoverDebit,
        decimal TurnoverCredit,
        decimal OutgoingBalanceActive,
        decimal OutgoingBalancePassive,
        List<BillDTO> Bills);

    public record BillDTO(
        int BillNumber,
        decimal IncomingBalanceActive,
        decimal IncomingBalancePassive,
        decimal TurnoverDebit,
        decimal TurnoverCredit,
        decimal OutgoingBalanceActive,
        decimal OutgoingBalancePassive);
}
